pkg/v1/remote/catalog.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+)
+
+type catalog struct {
+	Repos []string `json:"repositories"`
+}
+
+// GetCatalogPage calls /_catalog, returning the list of repositories on the registry
+func GetCatalogPage(target name.Registry, last string, n int, options ...Option) ([]string, error) {
```

Comment:
I think the idea is to just make this a private method in the crane/catalog.go file. At some point in the future it can get moved to the remote library when we have interface concensus.

Reply:
I think I'm actually okay with leaving this in. If we ever come up with a good API for this, we can name it `GetCatalog`, and `GetCatalogPage` would still be useful.

I am a little bit unhappy with this because it doesn't allow clients to adhere to [the spec](https://docs.docker.com/registry/spec/api/#pagination):

> Compliant client implementations should always use the Link header value when proceeding through results linearly.

However, this does enable you to skip ahead, which is immediately after:

> The client may construct URLs to skip forward in the catalog.

I think we're basically making it impossible (sometimes, depends on implementation) for a registry to implement this efficiently by handing us a cursor in the Link header, but maybe we don't care.